src/libs/actions/Report.js

```diff
@@ -950,6 +950,14 @@ function fetchAllReports(
         .then((returnedReportIDs) => {
             Onyx.set(ONYXKEYS.INITIAL_REPORT_DATA_LOADED, true);
 
+            // If at this point the user still doesn't have a Concierge report, create it for them.
+            // This means they were a participant in reports before their account was created (e.g. default rooms)
+            const hasConciergeChat = _.some(allReports, report => isConciergeChatReport(report));
```

Comment:
Couple questions about this...

1. Can we guarantee that the local variable of `allReports` has all the reports that we set here?

https://github.com/Expensify/App/blob/100cc8c3ca71ab76f80b9c9abcf720b6f325c865/src/libs/actions/Report.js#L378

Because I'm pretty sure it won't have them yet if the expectation is that they were set in that promise and we can now access them here.

2. If we expect one of the reports in `allReports` to be a concierge report then we should already know this based on the logic here:

https://github.com/Expensify/App/blob/100cc8c3ca71ab76f80b9c9abcf720b6f325c865/src/libs/actions/Report.js#L1243-L1247

And in that case can we just check to see if we have a `conciergeChatReportID`?

I think the answer to my second question is that it still wouldn't work because we first need to fix the problem with my first question.

Reply:
1. Are you saying that the promise from `fetchChatReportsByIDs` won't be resolved yet? Or are you saying that the Onyx callback won't have run yet? 
  - Looking at the logic the promise returned from `fetchChatReportsByIDs` should be resolved by the time we access `allReports` here. The only thing I can see is that we update `allReports` in an Onyx callback that should get triggered while we're resolving `fetchChatReportsByIDs`, but the callback may not actually be invoked till after we access `allReports` here. Is that your concern?

Reply:
>Or are you saying that the Onyx callback won't have run yet?

👍 

>the callback may not actually be invoked till after we access allReports here. Is that your concern?

👍